src/tagish/src/main/java/com/tagish/auth/DBLogin.java

```diff
@@ -114,28 +139,66 @@ protected synchronized Vector validateUser(String username, char password[]) thr
 		}
 	}
 
+	private int getFailedLoginCount(Connection con, String username, AuditEventType event) {
+
+		ResultSet rsu = null, rsr = null;
+		PreparedStatement psu = null;
+		int failureCount = 0;
+
+		try {
+			psu = con.prepareStatement("SELECT " + eventTypeColumn +
+										" FROM " + auditTable +
+										" WHERE " + principalIdColumn + "=?" +
+										" AND " + eventDateColumn + ">= ?");
+

```

Comment:
Yep, because we'll need to figure out the most recent event.. We're missing a key bit, the lockout period.. which will need compared against the time of that event.